server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java

```diff
@@ -803,28 +718,12 @@ public void flushOneMemTable() {
             memTableToFlush.isSignalMemTable(), flushingMemTables.size());
       }
     }
+    Future<Void> vmFuture = VmMergeTaskPoolManager.getInstance().submit(new VmMergeLoop());
```

Comment:
copy the vmWriters and vmResources and set to VmMergeLoop to avoid being changed.